auth/options.go

```diff
@@ -0,0 +1,87 @@
+package auth
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type RemoteOAuthTokenOption func(*RemoteOAuthToken)
+
+func WithAccessToken(token, tokenType string, expiry time.Time) RemoteOAuthTokenOption {
+	return func(t *RemoteOAuthToken) {
+		t.AccessToken = token
+		t.TokenType = tokenType
+		t.Expiry = expiry
+	}
+}
+
+func WithCloudEnv() RemoteOAuthTokenOption {
```

Comment:
This can be the default/implicit as it's the only reason to use this package and won't be enabled unless `CQ_CLOUD` env var exists. If we did that some of the ID/UUID confusion and post-opt-validate code can be simplified.

Reply:
Done in https://github.com/cloudquery/plugin-sdk/pull/1866/commits/66d2cabdf8385d4e92b9e1de7f6f9281a12a3d70